control-plane/agents/src/bin/core/controller/registry.rs

```diff
@@ -186,9 +188,26 @@ impl Registry {
         config
     }
 
+    /// Disable the v1 compat mode, once all nexus info keys are migrated.
+    pub(crate) async fn disable_v1_compat<S: Store>(store: &mut S) -> CoreRegistryConfig {
+        let config = CoreRegistryConfig::new(NodeRegistration::Automatic);
```

Comment:
Which should be fine right we don't have any field expect v1 compat that changes.

Reply:
yes we do... you can modify etcd directly.. also let's make it ready for the future when we do expose the knobs to modify it, otherwise this will very silently corrupt the config.
Also this should take self Registry so we can also modify the in-memory config.

Reply:
Sure. I will take the config from self and just modify the v1 compat and update accordingly. Thanks